### DISCUSSION TITLE
Clickhouse Asynchronous I/O

### DISCUSSION THREAD

**Discussion by ArijitSinghEDA** [sqlalchemy/sqlalchemy#10754] [Usage Questions] [answered]
I have a sql query: `SELECT * FROM TABLE1`
The table is stored in a Clickhouse database.

I am running the following code:
```
from sqlalchemy import text
from sqlalchemy.ext.asyncio import create_async_engine

engine = create_async_engine("clickhouse+asynch://host/db")

async with engine.connect() as conn:
    res = await conn.stream(text(query))
    data = await res.partitions(10000)
```

This results into the following error: 
```
asynch.errors.ServerException: Code: 12.
DB::Exception: Setting 'max_block_size' cannot be zero
```

What am I doing wrong? My objective is to get data in parallel

**Comment by CaselIT** [comment, marked as answer]
Hi,

You would have to ask to the clickhouse driver developers. That's an external package not maintained here

**Reply by ArijitSinghEDA** [reply]
Okay. Thank you for the response
